### PR TITLE
Serve registry mirrors as salt expects it

### PR DESCRIPTION
**Pull request by ereslibre**

### CODE REVIEW

File: app/controllers/internal_api/v1/pillars_controller.rb [hunk 1]
```diff
@@ -20,13 +20,22 @@ def pillar_contents
...
-        mirror:      reg.mirror,
-        certificate: (reg.certificate.present? ? reg.certificate.certificate : nil)
-      }
+    registries = DockerRegistry.is_registry.map do |reg|
+      url:  reg.url,
+      cert: (reg.certificate.present? ? reg.certificate.certificate : nil)
     end
-    { registries: registries }
+    registry_mirrors = []
+    DockerRegistry.is_mirror.group(:mirror).pluck(:mirror).each do |mirror|
```
Comment (vitoravelino) [inline app/controllers/internal_api/v1/pillars_controller.rb]: Rename `|mirror|` to something like  `|remote_registry|` just to make it clear that is the url of a registry and not a mirror.

**Comment by ereslibre** [comment]
Needs testing.

Supersedes: https://github.com/kubic-project/velum/pull/414

**Comment by MaximilianMeister** [comment]
@ereslibre tested, looks good

```bash
caasp-admin:~ # docker exec -it $(docker ps | grep "salt-master_velum" | awk '{print $1}') salt "2e3c02a5a02a414a9b3eda270528bb40" cmd.run "cat /etc/docker/daemon.json"
2e3c02a5a02a414a9b3eda270528bb40:
    {
      "registries": [
        {
          "Mirrors": [
            {
              "URL": "https://docker-testing-registry.suse.de"
            }
          ],
          "Prefix": "https://registry.suse.com"
        }
      ],
      "iptables":false,
      "log-level": "warn"
    }
```

```bash
caasp-admin:~ # docker exec -it $(docker ps | grep "salt-master_velum" | awk '{print $1}') salt "2e3c02a5a02a414a9b3eda270528bb40" cmd.run "cat /etc/docker/certs.d/docker-testing-registry.suse.de/ca.crt"
2e3c02a5a02a414a9b3eda270528bb40:
    Certificate:
        Data:
            Version: 3 (0x2)
            Serial Number:
  ...
```

**Comment by ereslibre** [comment]
[salt: configure the docker engine to use on-premise registry mirrors](https://trello.com/c/skVwgvED/478-salt-configure-the-docker-engine-to-use-on-premise-registry-mirrors)

**Pull request merged**